integrations/access/accessrequest/request_mapping.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accessrequest
+
+import (
+	"time"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/expression"
+	"github.com/gravitational/teleport/lib/utils/typical"
+	"github.com/gravitational/trace"
+)
+
+const (
+	timeFormat = time.RFC3339 // "2006-01-02T15:04:05Z07:00"
+)
+
+// accessRequestExpressionEnv holds user details that can be mapped in an
+// access request condition assertion.
+type accessRequestExpressionEnv struct {
+	Roles              []string
+	SuggestedReviewers []string
+	Annotations        map[string][]string
+	User               string
+	RequestReason      string
+	CreationTime       time.Time
+	Expiry             time.Time
+}
+
+type accessRequestExpression typical.Expression[accessRequestExpressionEnv, any]
+
+func parseAccessRequestExpression(expr string) (accessRequestExpression, error) {
+	parser, err := newRequestConditionParser()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	parsedExpr, err := parser.Parse(expr)
+	if err != nil {
+		return nil, trace.Wrap(err, "parsing label expression")
```

Comment:
Is this actually a label expression, or was this copy pasted from other code and never updated?

Label expressions are part of Teleport's RBAC system. I think this is a separate concept.

Reply:
Not sure I ever saw an answer to this question.

Reply:
Have fixed this now.